tests/momento/simple_cache_client/shared_behaviors.py

```diff
@@ -10,76 +11,97 @@
 from momento.typing import TCacheName, TScalarKey
 from tests.utils import uuid_str
 
-TCacheNameValidator = Callable[[TCacheName], CacheResponse]
+
+class TCacheNameValidator(Protocol):
+    def __call__(self, cache_name: TCacheName) -> CacheResponse:
+        ...
 
 
 def a_cache_name_validator() -> None:
     def with_non_existent_cache_name_it_throws_not_found(cache_name_validator: TCacheNameValidator) -> None:
         cache_name = uuid_str()
-        response = cache_name_validator(cache_name)
-        assert isinstance(response, ErrorResponseMixin)
-        assert response.error_code == MomentoErrorCode.NOT_FOUND_ERROR
+        response = cache_name_validator(cache_name=cache_name)
+        if isinstance(response, ErrorResponseMixin):
+            assert response.error_code == MomentoErrorCode.NOT_FOUND_ERROR
+        else:
+            assert False
```

Comment:
I don't understand this refactor (or the others that are like this)  :thinking:  Is there a typing issue, or something else? The previous version will give better detail when the assert fails and is also shorter

Reply:
Yes, without this `response` is a generic response and there's a type error. This uses type narrowing to solve that. There were so many type errors like this it was hard to work with on other type errors.

This happens so often a custom assert would be a better solution, but this fixes the type error.

Reply:
I recommend just adding the assert back above the if statement and remove the else
```suggestion
        assert isinstance(response, ErrorResponseMixin)
        if isinstance(response, ErrorResponseMixin):
            assert response.error_code == MomentoErrorCode.NOT_FOUND_ERROR
```

Reply:
I'd rather not have to repeat that sort of code everywhere. A little `def assert_response_is_error(response) -> ErrorResponseMixin:`  would solve the problem.

Reply:
I pushed a change which introduces a custom assert to handle this. I can't spend much time on it so I only applied it to the shared behaviors. The rest should be a SMOP.

Reply:
TIL "SMOP" :smile: